docs/sql-keywords.md

```diff
@@ -280,6 +280,7 @@ Below is a list of all the keywords in Spark SQL.
   <tr><td>UNKNOWN</td><td>reserved</td><td>non-reserved</td><td>reserved</td></tr>
   <tr><td>UNLOCK</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
   <tr><td>UNSET</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
+  <tr><td>UPDATE</td><td>non-reserved</td><td>non-reserved</td><td>reserved</td></tr>
```

Comment:
if UPDATE is reserved in SQL spec, we should make it reserved when ansi parser is enabled.

We should put UPDATE in `ansiReserved`.

Reply:
cc @maropu

Reply:
This behavior is the same with DELETE, which is non-reserved in both the ansi and default mode in sparksql, but reserved in SQL spec 2011. I checked it in postgresql and mysql, both DELETE and UPDATE are non-reserved, but in Mysql they are reserved.

Reply:
I checked other keywords and seems like the general rule here is to follow Postgres, we can revisit it later. cc @gengliangwang